source-dropbox/main.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"cmp"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"slices"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox/files"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+type config struct {
+	Path        string         `json:"path"`
+	Credentials *Credentials   `json:"credentials"`
```

Comment:
Credentials here doesn't need to be a pointer. Parser doesn't really either but I know we do that elsewhere so that's fine, and I guess it makes the `ParserConfig()` method work arguably a little cleaner.